pkg/server/server.go

```diff
@@ -99,7 +99,8 @@ type Server struct {
 	experimentals                    []ExperimentalFeatureFlag
 	serviceName                      string
 
-	typesystemResolver typesystem.TypesystemResolverFunc
+	typesystemResolver     typesystem.TypesystemResolverFunc
+	typesystemResolverStop func()
```

Comment:
this doesn't look super nice to me but it's the fastest way to remove the leak. We can make this nicer later

Reply:
Makes sense. I think it's fine. Nice addition 👍